### DISCUSSION TITLE
If the PK of an Auditable entity is set by the database, can I audit an insert which will show the value of this key?

### DISCUSSION THREAD

**Discussion by marazarian** [thepirat000/Audit.NET#646] [Q&A]
Using an SQL database and EF8, if I have a customer table where the PK, customerId , is set by the database, when my code calls saveChangesAsync() to insert a new customer , at that time I don't have the customerId number  but only a temporary key set by EF. My audit table should track the new insert including the new customerId number. Would Audit.Net allow this?

**Comment by thepirat000** [comment]
Yes, by default you will get the database generated ID in the `PrimaryKey` and `ColumnValues` properties of the audit event, for example:

![image](https://github.com/thepirat000/Audit.NET/assets/9836380/8fa15015-bd2c-4f7c-8821-3f396e8768d2)
